cmd/goal/clerk.go

```diff
@@ -97,6 +98,69 @@ var clerkCmd = &cobra.Command{
 	},
 }
 
+func waitForCommit(client libgoal.Client, txid string) {
+	if noWaitAfterSend {
+		return
+	}
+
+	// Get current round information
+	stat, err := client.Status()
+	if err != nil {
+		reportErrorf(errorRequestFail, err)
+	}
+
+	for {
+		// Check if we know about the transaction yet
+		txn, err := client.PendingTransactionInformation(txid)
+		if err != nil {
+			reportErrorf(errorRequestFail, err)
+		}
+
+		if txn.ConfirmedRound > 0 {
+			reportInfof(infoTxCommitted, txid, txn.ConfirmedRound)
+			break
+		}
+
+		if txn.PoolError != "" {
+			reportErrorf(txPoolError, txid, txn.PoolError)
```

Comment:
You changed the contact of changeAccountOnlineStatus() -- you need to return the error and propagate it.  generateAndRegisterPartKey() for example needs this to clean up in case of error.

Reply:
Oops, missed this difference.  Thanks!